src/main/java/view/SellView.java

```diff
@@ -155,26 +173,36 @@ public void propertyChange(PropertyChangeEvent evt) {
         sellViewModel.setState(state);
 
         // Get owned stocks list from dashboard for dropdown menu on sell page
-        DashboardState dashboardState = dashboardViewModel.getState();
-        List<String> ownedStocks = dashboardState.getOwnedTickers();
+        List<String> ownedStocks = state.getOwnedStocks();
         stockInputField.removeAllItems();
         if (ownedStocks != null && !ownedStocks.isEmpty()) {
             for (String s: ownedStocks) {
                 stockInputField.addItem(s);
             }
         }
 
+        List<Double> ownedAmounts = state.getOwnedAmounts();
+        List<Double> sellPrices = state.getSellAmounts();
+
+        DefaultTableModel tableModel = new DefaultTableModel();
+        tableModel.addColumn("Stock");
+        tableModel.addColumn("Amount Owned");
+        tableModel.addColumn("Sell Price (Single Unit)");
+        tableModel.addColumn("Sell Price (All Units)");
+
         // Show table of stocks and amounts owned
-        HashMap<String, String> ownedStocksAmounts = new HashMap<String, String>();
-        List<Double> ownedAmounts = dashboardState.getOwnedAmounts();
-        if (ownedStocks != null && ownedAmounts != null && !ownedAmounts.isEmpty() && !ownedStocks.isEmpty()) {
-            for (int i = 0; i<ownedStocks.size(); i++) {
-                ownedStocksAmounts.put(ownedStocks.get(i), String.valueOf(ownedAmounts.get(i)));
-            }
+        for (int i = 0; i < ownedStocks.size(); i++) {
+            String stockTicker = ownedStocks.get(i);
+            Double amountOwned = ownedAmounts.get(i);
+            Double sellPriceSingle = sellPrices.get(i);
+            Double sellPriceAll = sellPriceSingle * amountOwned;
+            tableModel.addRow(new Object[] {stockTicker, amountOwned, sellPriceSingle, sellPriceAll});
```

Comment:
nice efficiency improvement

Reply:
its the same runtime/number of iterations as before, I dont see whats wrong with it.